nms/app/components/CustomMetrics.tsx

```diff
@@ -49,25 +51,29 @@ export function getStep(
 // hence we have to split the start and end window into several sets of
 // [start, end] queries which can then be queried in parallel
 export function getQueryRanges(
-  start: moment.Moment,
-  end: moment.Moment,
+  start: Date,
+  end: Date,
   delta: number,
   unit: TimeUnit,
-): Array<[moment.Moment, moment.Moment]> {
-  const queries: Array<[moment.Moment, moment.Moment]> = [];
-  let s = start.clone();
-  // go back delta time so that we get the total number of events
-  // or logs at that 's' point of time
-  s = s.subtract(delta, unit);
-  while (end.diff(s, unit) >= delta) {
-    const e = s.clone();
-    e.add(delta, unit);
-    queries.push([s, e]);
-    s = e;
+): Array<[Date, Date]> {
+  const queries: Array<[Date, Date]> = [];
+  let intervalStart = toDate(start);
+  while (isBefore(intervalStart, end)) {
+    const intervalEnd = add(intervalStart, {[timeUnitToDuration(unit)]: delta});
+    queries.push([intervalStart, intervalEnd]);
+    intervalStart = intervalEnd;
   }
   return queries;
 }
 
+function timeUnitToDuration(unit: TimeUnit): keyof Duration {
+  if (unit === 'millisecond' || unit === 'quarter') {
+    throw new Error(`${unit} cannot be converted to Duration!`);
+  } else {
+    return `${unit}s`;
+  }
+}
+
```

Comment:
Might not be the best solution, however, this adapter of mismatching types is the easiest way to solve the issue for now.
I wonder, why the Chart.js adapter for date-fns cannot do such things, or why it is not clearly documented how this should be done. :man_shrugging: